os/drivers/video/video_null_driver.c

```diff
@@ -0,0 +1,1554 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <semaphore.h>
+#include <assert.h>
+#include <errno.h>
+#include <debug.h>
+
+#include <tinyara/kmalloc.h>
+#include <tinyara/fs/fs.h>
+#include <tinyara/arch.h>
+#include <tinyara/wqueue.h>
+
+#include <video/video_halif.h>
+
+#include "video_null_driver.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+#define video_null_givesem(s)		sem_post(s);
+
+#define FILE_FORMAT "/mnt/%dx%d.%s"
+#define FILE_NAMELEN  64
+
+#define MAX_FRAMES 7
+/* Used only in case of discrete */
+#define MAX_INTERVAL 5
+
+#define CHECK_RANGE(value,min,max,step) do { \
+                                          if ((value < min) || \
+                                              (value > max) || \
+                                              ((value - min) % step != 0)) \
+                                            { \
+                                              return -EINVAL;\
+                                            } \
+                                         } while (0)
```

Comment:
You should not use spaces for indentation.

Reply:
Done